src/independent_mogp.jl

```diff
@@ -115,10 +117,98 @@ Posterior implementation for isotopic inputs, given diagonal Σy (OILMM).
 See AbstractGPs.jl API docs.
 """
 function AbstractGPs.posterior(
-    ft::IsotropicFiniteIndependentMOGP, y::AbstractVector{<:Real}
+    ft::IsotropicByOutputsFiniteIndependentMOGP, y::AbstractVector{<:Real}
 )
     finiteGPs = finite_gps(ft, ft.Σy[1])
     ys = collect(eachcol(reshape(y, (length(ft.x.x), :))))
     ind_posts = [AbstractGPs.posterior(fx, y_i) for (fx, y_i) in zip(finiteGPs, ys)]
     return independent_mogp(ind_posts)
 end
+
+# AbstractGPs APIs implementations for MOInputIsotopicByFeatures.
+
+const IsotropicByFeaturesFiniteIndependentMOGP = FiniteGP{
+    <:IndependentMOGP,<:MOInputIsotopicByFeatures,<:Diagonal{<:Real,<:Fill}
+}
+
+# Indices which, when applied to a vector ordered by features, will order it by outputs.
+function indices_which_reorder_features_to_outputs(
+    x::Union{MOInputIsotopicByOutputs,MOInputIsotopicByFeatures}
+)
+    return collect(vec(reshape(1:length(x), length(x.x), x.out_dim)'))
```

Comment:
So if `x` is a `MOInputIsotopicByOutputs` Should `x[indices_which_reorder_features_to_outputs(x)]` return it ordered by features and similarily if `x` is a `MOInputIsotopicByFeaturess` Should `x[indices_which_reorder_features_to_outputs(x)]` return it ordered by outputs? Testing it in the repl it appears to work the one way (ordered by outputs -> ordered by features) but not the other